### ISSUE TITLE
How much SLAP is too much?

### ISSUE THREAD

**Issue by uggi121** [nus-cs2103-AY1920S1/forum#51]
How much abstraction is too much? Adding an extra layer of abstraction has its costs and benefits. In most cost-benefit analyses of abstraction, benefits would be a clear winner. 

SLAP seems to be a good guideline in general. However, it is highly likely we reach a stage where further abstractions cause more harm than good. That is, after a point, the costs (refactoring and debugging) would outweigh the benefits. What is a good guideline to follow to ensure that we're adding another layer of abstraction that has enough information to be significant? Is there a quantitative measure, like the reduction in the LoC in the original method, to determine the 'impact'?

**Comment by damithc** [comment]
Good point. It is usually a judgement call; there is no quantitative formula. Will touch on this at this week's lecture.
